src/CaptureService/include/CaptureService/CaptureService.h

```diff
@@ -30,9 +30,24 @@ class CaptureService : public orbit_grpc_protos::CaptureService::Service {
   void RemoveCaptureStartStopListener(CaptureStartStopListener* listener);
 
  protected:
-  grpc::Status InitializeCapture(
-      grpc::ServerReaderWriter<orbit_grpc_protos::CaptureResponse,
-                               orbit_grpc_protos::CaptureRequest>* reader_writer);
+  enum class CaptureInitializationResult { kSuccess, kAlreadyInProgress };
+
+  class ClientCaptureEventCollectorBuilder {
+   public:
+    virtual ~ClientCaptureEventCollectorBuilder() = default;
+    virtual std::unique_ptr<orbit_producer_event_processor::GrpcClientCaptureEventCollector>
+    BuildClientCaptureEventCollector() = 0;
+  };
+
+  class StartStopCaptureRequestWaiter {
+   public:
+    virtual ~StartStopCaptureRequestWaiter() = default;
+    virtual orbit_grpc_protos::CaptureOptions WaitForStartCaptureRequest() = 0;
+    virtual void WaitForStopCaptureRequest() = 0;
+  };
```

Comment:
I was hoping we could avoid this. But you're right. This is probably the best option.